python/pyspark/pandas/series.py

```diff
@@ -7092,15 +7092,15 @@ def resample(
         ----------
         rule : str
             The offset string or object representing target conversion.
-            Currently, supported units are {'Y', 'A', 'M', 'D', 'H',
-            'T', 'MIN', 'S'}.
+            Currently, supported units are {'YE', 'A', 'ME', 'D', 'h',
+            'min', 'MIN', 's'}.
```

Comment:
Is this a breaking change?

Reply:
Yeah, Pandas 2.2.0 brings couple of breaking changes so we should make sure we ship this support after Spark 4.0.0.

See [related update from Pandas release note](https://pandas.pydata.org/docs/whatsnew/v2.2.0.html#deprecate-aliases-m-q-y-etc-in-favour-of-me-qe-ye-etc-for-offsets) for more detail.

Reply:
Ya, that comes to my second question. (https://github.com/apache/spark/pull/44881#pullrequestreview-1889581226).

Even for the users who choose old Pandas libraries, Apache Spark enforces this breaking change, @itholic ?

Reply:
The background of my question is that `Data Science` team has been struggling when they validates their pipelines on new Spark versions.

Reply:
> Even for the users who choose old Pandas libraries, Apache Spark enforces this breaking change

Yes. In current policy, if we want to use latest Apache Spark then we cannot avoid having to follow the behavior of latest Pandas as well IIRC.

Reply:
However, the current rule is that it should not be accompanied by such a breaking change unless the major version changes.

This means that users should be able to use their pipeline as is, as long as they are using at least version 3.x of Spark.

Reply:
Oh, sorry that was my mistake. This should work even in old Pandas before Spark 4.0.0 release.

Let me fix them to work both Pandas 2.2.0 and old Pandas.

Reply:
Just updated to resample work in old Pandas as well. Now it's safe.